src/box/gc.h

```diff
@@ -351,24 +367,28 @@ void
 gc_unref_checkpoint(struct gc_checkpoint_ref *ref);
 
 /**
- * Register a consumer.
+ * Register a consumer. It can be a newly allocated consumer, or a consumer
+ * loaded from persistent WAL GC state and waiting for registration.
+ * Only one consumer with given UUID can be registered at the same time.
  *
  * This will stop garbage collection of WAL files newer than
  * @vclock until the consumer is unregistered or advanced.
  * @type specifies a type of object the consumer belongs to,
  * @uuid specifies UUID of this object. They will be
- * used for listing the consumer in box.info.gc().
+ * used for listing the consumer in box.info.gc(), also UUID
+ * is used to ensure uniqueness of consumers.
  *
- * Returns a pointer to the new consumer object or NULL on
+ * Returns a pointer to the consumer object or NULL on
  * memory allocation failure.
```

Comment:
Nit: If it returns NULL only on memory allocation failure, it may be worth making it fail-free while we're at it by using xmalloc.

Reply:
Done, did it in a separate commit.